platform/core.multitabs/src/org/netbeans/core/multitabs/impl/ButtonPopupSwitcher.java

```diff
@@ -355,6 +354,22 @@ private boolean changeSelection( KeyEvent event ) {
                         selRow = Math.min( pTable.getModel().getRowCount()-1, selRow );
                         selCol = Math.min( pTable.getModel().getColumnCount()-1, selCol );
                         switched = true;
+                    }else if( tabIndex == -1 ){
+                        List<TabData> tabs = controller.getTabModel().getTabs();
+                        ProjectProxy project = item.getProject();
+                        ProjectSupport projectSupport = ProjectSupport.getDefault();
+                        for ( int i = tabs.size(); i-- > 0; ) {  //reverse iteration to close tabs from the end so tabIndex does not change
+                            if( controller.getTabModel().size() == 1 ) {
+                                SwingUtilities.invokeLater(this::hideCurrentPopup);
+                            }
+                            TabData projectTab = tabs.get( i );
+                            ProjectProxy projectForTab = projectSupport.getProjectForTab( projectTab );
+                            if (( project == null && projectForTab == null ) || ( projectForTab != null && projectForTab.equals( project ))) {
+                                tabIndex = controller.getTabModel().indexOf( projectTab );
+                                TabActionEvent tae = new TabActionEvent( this, TabbedContainer.COMMAND_CLOSE, tabIndex );
+                                controller.postActionEvent( tae );
+                            }
+                        }
```

Comment:
step 2, remove this. and call the method:
```
pTable.closeTabsOfProject(project);
```
I think project might be the only param you need.

Reply:
Ohh, I over looked the pTable variable. thank you. Will commit asap.

Reply:
awesome